lm_eval/models/ggml.py

```diff
@@ -0,0 +1,96 @@
+import requests
+import logging
+from lm_eval.base import BaseLM
+from tqdm import tqdm
+from requests.exceptions import RequestException
+
+logger = logging.getLogger(__name__)
+
+def ggml_completion(base_url, **kwargs):
+    try:
+        response = requests.post(f"{base_url}/v1/completions", json=kwargs)
+        response.raise_for_status()
+        return response.json()
+    except RequestException as e:
+        print(f"RequestException: {e}")
+        return None
+
+class GGMLLM(BaseLM):
+    def __init__(self, base_url, truncate=False):
+        super().__init__()
+        self.base_url = base_url
+        self.truncate = truncate
+
+    def loglikelihood(self, requests):
+        res = []
+        for context, continuation in tqdm(requests):
+            response = ggml_completion(self.base_url, context=context, continuation=continuation)
+            if response and "choices" in response and response["choices"]:
+                choice = response["choices"][0]
+                logprobs = choice.get("logprobs")
+                logprob = logprobs["token_logprobs"][0] if logprobs and logprobs["token_logprobs"] else -1.2345
```

Comment:
2 points on this: (apologies, I'm not familiar with the LLaMA-cpp interface):

1. what does the 0 index in `logprobs["token_logprobs"][0]` do? what sort of object is this and which tokens does `logprob` then store logprobs for?
2. This should raise an error if logprobs aren't returned by the model rather than failing silently.

Reply:
Added an error if logprobs aren't returned.